src/items.rs

```diff
@@ -1768,10 +1768,27 @@ fn rewrite_ty<R: Rewrite>(
 
     if let Some(bounds) = generic_bounds_opt {
         if !bounds.is_empty() {
-            // 2 = `: `
-            let shape = Shape::indented(indent, context.config).offset_left(result.len() + 2)?;
-            let type_bounds = bounds.rewrite(context, shape).map(|s| format!(": {}", s))?;
-            result.push_str(&type_bounds);
+            match context.config.style_edition() {
+                style_edition if style_edition < StyleEdition::Edition2024 => {
+                    // 2 = `: `
+                    let shape =
+                        Shape::indented(indent, context.config).offset_left(result.len() + 2)?;
+                    let type_bounds = bounds.rewrite(context, shape).map(|s| format!(": {}", s))?;
+                    result.push_str(&type_bounds);
+                }
+                _ => {
+                    let shape =
+                        Shape::indented(indent, context.config).offset_left(result.len())?;
+                    result = rewrite_assign_rhs_with(
+                        context,
+                        result.clone() + ":",
+                        bounds,
+                        shape,
+                        &RhsAssignKind::Bounds,
+                        RhsTactics::ForceNextLineWithoutIndent,
+                    )?;
+                }
+            }
```

Comment:
I don't think we need to use a match here. I would probably write this using an `if else` since you can avoid an extra level on indentation. Also, let's use `<= StyleEdition::Edition2021` for older formatting instead of `< StyleEdition::Edition2024`. It's easier to grep for `Edition2021` to find all the places where we're using the older formatting rules.